editor/import/scene_import_settings.cpp

```diff
@@ -850,6 +858,127 @@ void SceneImportSettings::_select(Tree *p_from, String p_type, String p_id) {
+		} else {
+			animation_player->play(id);
+			animation_play_button->set_icon(get_theme_icon(SNAME("Pause"), SNAME("EditorIcons")));
+			set_process(true);
+		}
+	}
+}
+
+void SceneImportSettings::_stop_current_animation() {
+	animation_pingpong = false;
+	animation_player->stop();
+	animation_play_button->set_icon(get_theme_icon(SNAME("MainPlay"), SNAME("EditorIcons")));
+	animation_slider->set_value_no_signal(0.0);
+	set_process(false);
+}
+
+void SceneImportSettings::_reset_animation(const String &p_animation_name) {
+	if (p_animation_name.is_empty()) {
+		animation_preview->hide();
+
+		if (animation_player != nullptr && animation_player->is_playing()) {
+			animation_player->stop();
+		}
+		animation_play_button->set_icon(get_theme_icon(SNAME("MainPlay"), SNAME("EditorIcons")));
+
+		_reset_bone_transforms();
+		set_process(false);
+	} else {
+		_reset_bone_transforms();
+		animation_preview->show();
+
+		animation_loop_mode = Animation::LoopMode::LOOP_NONE;
+		animation_pingpong = false;
+
+		if (animation_map.has(p_animation_name)) {
+			HashMap<StringName, Variant> settings = animation_map[p_animation_name].settings;
+			if (settings.has("settings/loop_mode")) {
+				animation_loop_mode = static_cast<Animation::LoopMode>((int)settings["settings/loop_mode"]);
+			}
+		}
+
+		if (animation_player->is_playing() && animation_loop_mode != Animation::LoopMode::LOOP_NONE) {
+			animation_player->play(p_animation_name);
+		} else {
+			animation_player->stop(true);
+			animation_play_button->set_icon(get_theme_icon(SNAME("MainPlay"), SNAME("EditorIcons")));
+			animation_player->set_assigned_animation(p_animation_name);
+			animation_player->seek(0.0, true);
+			animation_slider->set_value_no_signal(0.0);
+			set_process(false);
+		}
+	}
+}
+
+void SceneImportSettings::_animation_slider_value_changed(double p_value) {
+	if (animation_player == nullptr || !animation_map.has(selected_id) || animation_map[selected_id].animation.is_null()) {
+		return;
+	}
+	if (animation_player->is_playing()) {
+		animation_player->stop();
+		animation_play_button->set_icon(get_theme_icon(SNAME("MainPlay"), SNAME("EditorIcons")));
+		set_process(false);
+	}
+	animation_player->seek(p_value * animation_map[selected_id].animation->get_length(), true);
+}
+
+void SceneImportSettings::_animation_finished(const StringName &p_name) {
+	Animation::LoopMode loop_mode = animation_loop_mode;
+
+	switch (loop_mode) {
+		case Animation::LOOP_NONE: {
+			animation_play_button->set_icon(get_theme_icon(SNAME("MainPlay"), SNAME("EditorIcons")));
+			animation_slider->set_value_no_signal(1.0);
+			set_process(false);
+		} break;
+		case Animation::LOOP_LINEAR: {
+			animation_player->play(p_name);
+		} break;
+		case Animation::LOOP_PINGPONG: {
+			if (animation_pingpong) {
+				animation_player->play(p_name);
+			} else {
+				animation_player->play_backwards(p_name);
+			}
+			animation_pingpong = !animation_pingpong;
+		} break;
+		default: {
+		} break;
+	}
```

Comment:
It seems a bit redundant to have to write the processing for looping outside of the AnimationPlayer, but I think this is the only way to do it for now. I will consider providing a better API in a future refactoring planned as https://github.com/godotengine/godot-proposals/issues/5972.